File: 1143. longest common subsq/memory-op.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        len1, len2 = len(text1), len(text2)
        dp = [0] * (len2 + 1)
        for i in range(len1):
            cur = [0] * (len2 + 1)
            for j in range(len2):
                if text1[i] == text2[j]:
                    cur[j + 1] = 1 + dp[j]
                else:
                    cur[j + 1] = max(dp[j + 1], cur[j])
            dp = cur
        
        return dp[-1]

File: 1143. longest common subsq/memory-op.rs
impl Solution {
    pub fn longest_common_subsequence(text1: String, text2: String) -> i32 {
        let l1 = text1.len();
        let l2 = text2.len();
        let t1:Vec<char> = text1.chars().collect();
        let t2:Vec<char> = text2.chars().collect();
        let mut dp = vec![0; l2 + 1];

        for i in 0..l1{
            let mut cur = vec![0; l2 + 1];
            for j in 0..l2{
                if t1[i] == t2[j]{
                    cur[j + 1] = 1 + dp[j];
                }else{
                    cur[j + 1] = dp[j + 1].max(cur[j]);
                }
            }
            dp = cur;
        }
        dp[l2]
    }
}